### PR TITLE
Feature/add time selector

### PR DESCRIPTION
**Pull request by flibbertigibbet**
Adds extra options section to directions tab.
Adds mode and time selectors to directions, and uses them in planning trips.

### CODE REVIEW

**Comment by flibbertigibbet** [comment]
image

File: src/app/scripts/cac/pages/cac-pages-map.js [hunk 1]
```diff
@@ -57,8 +61,9 @@ CAC.Pages.Map = (function ($, Handlebars, _, MapControl, Routing, MockDestinatio
 
     return Map;
 
-    function showOptions() {
-        var moreOpt = '.sidebar-options .more-options';
+    function showOptions(event) {
+        var parent = $(event.target).parent().parent();
```
Comment (CloudNiner) [inline src/app/scripts/cac/pages/cac-pages-map.js]: I tend to strongly avoid selections like this, as it is extremely brittle, and I've been bitten by it before. At least with an id/class selector of some kind, you avoid breaking the selection if you move the dom tree around.

File: src/app/scripts/cac/pages/cac-pages-map.js [hunk 2]
```diff
@@ -57,8 +61,9 @@ CAC.Pages.Map = (function ($, Handlebars, _, MapControl, Routing, MockDestinatio
...
+        var moreOpt = $('.sidebar-options .more-options', parent);
 
         $(moreOpt).toggleClass('active');
         $(moreOpt).parent().find('a.view-more').text(function() {
```
Comment (CloudNiner) [inline src/app/scripts/cac/pages/cac-pages-map.js]: Hmm an alternative to these parent selectors would be to select some wrapper element that is higher up the tree, and select the relevant children of it with standard selectors.

**Comment by CloudNiner** [comment]
+1, but strongly consider switching up those selectors

**Comment by flibbertigibbet** [comment]
image

File: src/app/scripts/cac/pages/cac-pages-map.js (hunk 2 above)
Comment (flibbertigibbet) [inline src/app/scripts/cac/pages/cac-pages-map.js]: Yes, but how else do I find it relative to the clicked link?  It was toggling both when either tab's expansion was clicked, and I couldn't figure out why `proxy` didn't seem to switch the context.

**Comment by flibbertigibbet** [comment]
A thousand sorries, but I'm going to go ahead and merge this is with the icky `parent().parent()` selector.  I have to go now, and will be out for a while, or else I'd spend more time on it.

**Pull request merged**